keras_nlp/src/models/resnet/resnet_backbone_test.py

```diff
@@ -0,0 +1,99 @@
+# Copyright 2024 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import pytest
+from absl.testing import parameterized
+from keras import backend
+from keras import ops
+
+from keras_nlp.src.models.resnet.resnet_backbone import ResNetBackbone
+from keras_nlp.src.tests.test_case import TestCase
+
+
+class ResNetBackboneTest(TestCase):
+    def setUp(self):
+        self.init_kwargs = {
+            "stackwise_num_filters": [64, 64, 64],
+            "stackwise_num_blocks": [2, 2, 2],
+            "stackwise_num_strides": [1, 2, 2],
+            "include_rescaling": False,
+            "pooling": "avg",
+        }
+        self.input_size = (16, 16)
+        self.input_data = ops.ones((2, 16, 16, 3))
+
+    @parameterized.named_parameters(
+        ("v1_basic_channels_last", False, "basic_block", "channels_last"),
+        ("v1_basic_channels_first", False, "basic_block", "channels_first"),
+        ("v1_block_channels_last", False, "block", "channels_last"),
+        ("v2_basic_channels_last", True, "basic_block", "channels_last"),
+        ("v2_basic_channels_first", True, "basic_block", "channels_first"),
+        ("v2_block_channels_last", True, "block", "channels_last"),
+    )
+    def test_backbone_basics(self, preact, block_type, data_format):
+        if (
+            backend.backend() == "tensorflow"
```

Comment:
lets not set the backend. This way the tests will run on all backends

Reply:
but tensorflow cpu can't perform convolution with the `channels_first` format
I can change it to detect whether the gpu is available or not.

Reply:
Added a check for TF CPU in newly introduced `run_vision_backbone_test`